### PR TITLE
Allow setting the Zipkin LocalEndpoint per-span

### PR DESCRIPTION
**Pull request by ramonza**
This is needed for rare cases when the logical endpoint differs from
the process submitting the span, for example Istio.

### CODE REVIEW

File: exporter/zipkin/zipkin.go [hunk 1]
```diff
@@ -29,6 +33,11 @@ import (
 type Exporter struct {
 	reporter      reporter.Reporter
 	localEndpoint *model.Endpoint
+
+	// Logger will be used to report errors from this exporter.
+	Logger interface {
```
Comment (rakyll) [inline exporter/zipkin/zipkin.go]: Can we instead have hook like onError at https://godoc.org/contrib.go.opencensus.io/exporter/stackdriver#Options?

It is useful for those who wants to report the errors to a service.

File: exporter/zipkin/zipkin_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package zipkin_test
```
Comment (rakyll) [inline exporter/zipkin/zipkin_integration_test.go]: Missing license header.

File: exporter/zipkin/zipkin_test.go [hunk 3]
```diff
@@ -18,11 +18,15 @@ import (
 	"encoding/json"
 	"io/ioutil"
 	"net/http"
-	"reflect"
 	"strings"
 	"testing"
 	"time"
```
Comment (rakyll) [inline exporter/zipkin/zipkin_test.go]: Remove the line and gofmt.

File: trace/basetypes.go [hunk 4]
```diff
@@ -50,16 +50,19 @@ type Attribute struct {
 }
 
 // BoolAttribute returns a bool-valued attribute.
+// Attribute keys starting with "opencensus" are reserved for internal use.
```
Comment (rakyll) [inline trace/basetypes.go]: Is this in the trace specs?

File: exporter/zipkin/zipkin.go [hunk 5]
```diff
@@ -38,25 +47,50 @@ type Exporter struct {
 // can be created with the openzipkin library, using one of the packages under
 // github.com/openzipkin/zipkin-go/reporter.
 //
-// localEndpoint sets the local endpoint of exported spans.  It can be
+// localEndpoint sets the default local endpoint of exported spans.  It can be
 // constructed with github.com/openzipkin/zipkin-go.NewEndpoint, e.g.:
 // 	localEndpoint, err := NewEndpoint("my server", listener.Addr().String())
 // localEndpoint can be nil.
+//
+// localEndpoint can be overridden on a per-span basis by setting Span attributes
```
Comment (rakyll) [inline exporter/zipkin/zipkin.go]: This doesn't look like a Zipkin-only case. Can't we implement it more generally?

We can export a canonical attribute key from the trace package for the service name.

Most importantly, it needs to be reflected in the specs especially if we will preserve the opencensus prefix to OpenCensus libraries.

**Comment by ramonza** [comment]
Going to take a different approach with Istio

**Pull request closed without merging**